loki/bulk/scheduler.py

```diff
@@ -709,6 +714,9 @@ def write_cmake_plan(self, filepath, mode, buildpath, rootpath):
         sources_to_transform = []
 
         for item in self.items:
+            if item.ignored:
```

Comment:
Coverage report suggests we don't have a test for the CMake plan... or am I just overlooking this somehow? Is this something we could add here maybe, also to have a baseline against which to check the new Scheduler?

Reply:
Hate to disagree with CodeCov, but...
https://github.com/ecmwf-ifs/loki/blob/main/tests/test_scheduler.py#L860

Maybe it's not recognising it because we're testing the effect somewhat indirectly - as in reading the generated file and checking that?

Reply:
I have to side with Codecov on that one...
The test uses the CMakePlanner, which writes the cmake file. Could it be the case that we're not actually using that function from the `Scheduler` class anymore at all? (In which case, it should definitely be deleted!)

Reply:
OMG, that does indeed look like an outdated test and needless code duplication. Will fix...